src/base/bittorrent/sessionimpl.h

```diff
@@ -811,6 +811,7 @@ namespace BitTorrent
         QMap<QString, CategoryOptions> m_categories;
         TagSet m_tags;
 
+        std::vector<lt::alert *> m_alerts;  // make it a class variable so it can preserve its allocated `capacity`
```

Comment:
Note that the alerts flow is irregular. Moreover, usually when we start the application, we have a lot of alerts at a time, and then the flow can weaken significantly. What's the point of constantly having a huge buffer to serve 2-3 items?

Reply:
>and then the flow can weaken significantly. 

Doesn't exit qbt also generate quite some alerts?

>What's the point of constantly having a huge buffer to serve 2-3 items?

Is this statistic gathered from heavy user or it is only on your machine?
Even if the alerts are few, it doesn't hurt to reuse the allocated memory as it can lessen memory fragmentation.

Reply:
>IMO, 1024 is good enough choice. At least it's better than default of 1.

Done.